src/main/java/ro/developmentfactory/thedrones/service/DronePositionServiceImpl.java

```diff
@@ -0,0 +1,146 @@
+package ro.developmentfactory.thedrones.service;
+
+import jakarta.persistence.EntityNotFoundException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import ro.developmentfactory.thedrones.controller.dto.DroneStatusResponse;
+import ro.developmentfactory.thedrones.repository.DroneRepository;
+import ro.developmentfactory.thedrones.repository.DroneStatusRepository;
+import ro.developmentfactory.thedrones.repository.entity.Direction;
+import ro.developmentfactory.thedrones.repository.entity.Drone;
+import ro.developmentfactory.thedrones.repository.entity.DroneStatus;
+
+import java.util.UUID;
+
+@Service
+public class DronePositionServiceImpl implements DronePositionService {
+
+    private final DroneStatusRepository droneStatusRepository;
+    private final DroneRepository droneRepository;
+    private static final Logger log = LoggerFactory.getLogger(DronePositionServiceImpl.class);
+
+    public DronePositionServiceImpl(DroneStatusRepository droneStatusRepository, DroneRepository droneRepository) {
+        this.droneStatusRepository = droneStatusRepository;
+        this.droneRepository = droneRepository;
+    }
+
+    @Override
+    @Transactional
+    public DroneStatusResponse turningDirection(DroneStatusResponse droneStatusResponse) {
+        UUID idDrone = droneStatusResponse.getIdDrone();
+
+        Drone drone = droneRepository.findById(idDrone)
+                .orElseThrow(() -> new EntityNotFoundException("Drone with id " + idDrone + " not found"));
+
+        if (droneStatusResponse.getTurnDirection() == null) {
+            throw new IllegalArgumentException("Turn direction cannot be null");
+        }
```

Comment:
if you need to do any validation, better do them first but then all reads, calculations and other operations